### PR TITLE
[IOTDB-2938]Some improvements and bug fix for New Sync

### PR DESCRIPTION
**Pull request by yschengzi**
1.refactor IoTDBSyncSenderIT framework, now it supports IT for deletions and IoTDB restarts.
2.add close in TsFilePipe for shutdown in SenderService.
3.fix bus when delete storage group
4.change SyncPathUitl#getSenderHistoryPipeDataDir to SyncPathUitl#getSenderHistoryPipeLogDir, change SyncPathUitl#getSenderRealTimePipeDataDir to SyncPathUitl#getSenderRealTimePipeLogDir.

### CODE REVIEW

File: integration/src/test/java/org/apache/iotdb/db/integration/sync/IoTDBSyncSenderIT.java [hunk 1]
```diff
@@ -276,18 +377,186 @@ public void testStopAndStartInsert() {
   @Test
   public void testRealTimeSchemaAndStopInsert() {
```
Comment (Cpaulyz) [inline integration/src/test/java/org/apache/iotdb/db/integration/sync/IoTDBSyncSenderIT.java:378]:
```suggestion
  public void testRealTimeAndStopInsert() {
```

File: server/src/main/java/org/apache/iotdb/db/sync/pipedata/queue/BufferedPipeDataQueue.java [hunk 2]
```diff
@@ -428,7 +434,7 @@ public static List<PipeData> parsePipeLog(File file) throws IOException {
         pipeData.add(PipeData.deserialize(inputStream));
       }
     } catch (EOFException e) {
-      logger.info(String.format("Finish parsing pipeLog %s.", file.getPath()));
+      // logger.info(String.format("Finish parsing pipeLog %s.", file.getPath()));
```
Comment (Cpaulyz) [inline server/src/main/java/org/apache/iotdb/db/sync/pipedata/queue/BufferedPipeDataQueue.java:437]: delete if unnecessary

File: server/src/main/java/org/apache/iotdb/db/sync/conf/SyncPathUtil.java [hunk 3]
```diff
@@ -39,13 +39,13 @@ public static String getSenderPipeDir(String pipeName, long createTime) {
         + String.format("-%s-%d", pipeName, createTime);
   }
 
-  public static String getSenderHistoryPipeDataDir(String pipeName, long createTime) {
+  public static String getSenderHistoryPipeLogDir(String pipeName, long createTime) {
     return getSenderPipeDir(pipeName, createTime)
```
Comment (Cpaulyz) [inline server/src/main/java/org/apache/iotdb/db/sync/conf/SyncPathUtil.java:43]: make directory structure on the sender side same as on the receiver side

File: server/src/main/java/org/apache/iotdb/db/sync/pipedata/queue/BufferedPipeDataQueue.java (hunk 2 above)
Comment (yschengzi) [reply server/src/main/java/org/apache/iotdb/db/sync/pipedata/queue/BufferedPipeDataQueue.java:437]: done

File: server/src/main/java/org/apache/iotdb/db/sync/conf/SyncPathUtil.java (hunk 3 above)
Comment (yschengzi) [reply server/src/main/java/org/apache/iotdb/db/sync/conf/SyncPathUtil.java:43]: done

File: integration/src/test/java/org/apache/iotdb/db/integration/sync/IoTDBSyncSenderIT.java [hunk 4]
```diff
@@ -274,20 +375,188 @@ public void testStopAndStartInsert() {
...
+      prepareIns3();
       startPipe();
+      prepareDel2();
+      prepareDel3();
+      stopPipe();
+
+      Thread.sleep(1000L); // check
+      //      for (PipeData pipeData : transportClient.getPipeDataList()) {
+      //        System.out.println(pipeData);
+      //      }
```
Comment (HTHou) [inline integration/src/test/java/org/apache/iotdb/db/integration/sync/IoTDBSyncSenderIT.java:461]: delete it?

Comment (yschengzi) [reply integration/src/test/java/org/apache/iotdb/db/integration/sync/IoTDBSyncSenderIT.java:461]: okk

**Pull request merged**